### PR TITLE
[stable/kong] Add imagePullSecrets Option

### PR DESCRIPTION
**Pull request by cilindrox**
Adds an `imagePullSecrets` option to enable private Docker registry usage.

### CODE REVIEW

**Comment by cilindrox** [comment]
pinging @shashiranjan84 

Also, I can bump the default image to the newer 0.14.1 if you'd like

**Comment by shashiranjan84** [comment]
@cilindrox thank you for PR. I am working on 0.14.1 version bump and I think it would cleaner if we do it as separate PR. I will review it ASAP.

File: stable/kong/templates/deployment.yaml [hunk 1]
```diff
@@ -23,6 +23,12 @@ spec:
         app: {{ template "kong.name" . }}
         release: {{ .Release.Name }}
     spec:
+      {{- if .Values.image.pullSecrets }}
```
Comment (shashiranjan84) [inline stable/kong/templates/deployment.yaml:29]: You would also need to update migration job.

Comment (cilindrox) [reply stable/kong/templates/deployment.yaml:29]: thanks for the quick turnaround. Updated the migrations template with `imagePullSecrets`.

**Comment by cilindrox** [comment]
@shashiranjan84 getting a `role "kong" does not exist` error. I'm assuming the RBAC is not being created properly on the test cluster?

Any ideas as to what could be going on?

**Comment by shashiranjan84** [comment]
@cilindrox, even my PR is failing with same error. Seems like some recent changes has been made. I did not get time to figure it out yet.

**Comment by cilindrox** [comment]
alright, lmk if I can help out with anything

**Comment by shashiranjan84** [comment]
> @shashiranjan84 getting a role "kong" does not exist error. I'm assuming the RBAC is not being created properly on the test cluster?

@cilindrox I don't think above is the root cause of test failure. Eventually role does get created and it been always like that.

@davidkarlsen any clue why build failing?

**Comment by hbagdi** [comment]
@cilindrox Mind rebasing this on top of master? The CI issue is fixed.

**Comment by cilindrox** [comment]
@hbagdi rebased, tests passing now. Should be g2g.

cc @shashiranjan84

**Comment by hbagdi** [comment]
@mattfarina @cpanato @davidkarlsen Can we merge this in with 0.5.1 version?
The chart version was recently bumped in another PR which got merged.

**Comment by hbagdi** [comment]
@unguiculus could help out in merge probably.
I merely help maintain the Kong chart.

**Comment by hbagdi** [comment]
#8768 just made it in, could you bump up the chart version one last time? We will merge it in then.

**Comment by hbagdi** [comment]
@cilindrox Also, please rebase/merge in master branch

**Comment by cilindrox** [comment]
@hbagdi rebased :feelsgood:

**Comment by hbagdi** [comment]
@cilindrox 

> @hbagdi rebased :feelsgood:


You are gonna have to 😡 once more. Too many PRs at the same time are leading to these version bump issues. I wish I could do this myself rather than asking you to do it but not authorized to do so.

**Comment by cilindrox** [comment]
@hbagdi re-re-re-re-rebased :P

**Pull request merged**